[iOS] Add templated cells as children of the CollectionView2

rmarinho opened the pull request:
### Description of Change

When porting to the new implementation we forgot to move add the inflated templates as a children of the ItemsView.

This pull request includes several changes to improve the functionality and maintainability of the `ItemsViewController2` and `TemplatedCell2` classes, as well as adding a new test case for a specific issue.  Also does some cleanup

Codebase simplification:

* [`src/Controls/src/Core/Handlers/Items2/iOS/ItemsViewController2.cs`](diffhunk://#diff-dcaa555cd62425342259680af963c19dc908ab24491c229d751caf3c056eeca7L214-L290): Removed unused methods `InvalidateMeasureIfContentSizeChanged`, `GetSize`, and `ConstrainItemsToBounds`. Also removed a debug `Console.WriteLine` statement from `DetermineCellReuseId` and updated `UpdateView` to properly disconnect handlers and dispose of views. [[1]](diffhunk://#diff-dcaa555cd62425342259680af963c19dc908ab24491c229d751caf3c056eeca7L214-L290) [[2]](diffhunk://#diff-dcaa555cd62425342259680af963c19dc908ab24491c229d751caf3c056eeca7L373) [[3]](diffhunk://#diff-dcaa555cd62425342259680af963c19dc908ab24491c229d751caf3c056eeca7L411-R334)

Enhancements to binding logic:

* [`src/Controls/src/Core/Handlers/Items2/iOS/TemplatedCell2.cs`](diffhunk://#diff-617eb1f606ac3f05750380c858c3fbef7b80cba5fc556bc3f53bf6b792380978R67): Updated the `Unbind` method to remove logical children and refactored the `Bind` method to use a helper method `BindVirtualView` for better code organization and clarity. [[1]](diffhunk://#diff-617eb1f606ac3f05750380c858c3fbef7b80cba5fc556bc3f53bf6b792380978R67) [[2]](diffhunk://#diff-617eb1f606ac3f05750380c858c3fbef7b80cba5fc556bc3f53bf6b792380978L109-R145)

New test case:

* `src/Controls/tests/TestCases.HostApp/Issues/Issue26066.xaml` and `src/Controls/tests/TestCases.HostApp/Issues/Issue26066.xaml.cs`: Added a new test case to address Issue 26066, which involves testing the `CollectionViewHandler2` with `RelativeSource` binding to `AncestorType`. [[1]](diffhunk://#diff-efb8508abc878ed9dee7cfc5fc5948ed37c969105aa83a9b1f892af2f825b617R1-R42) [[2]](diffhunk://#diff-84bb7f309b1e9ddf195f354c915e1e80d073d5d06ed60c8313796fc29570fb18R1-R42)
* [`src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue26066.cs`](diffhunk://#diff-35e2cfc839047627602556d9a99b3e6be67b9b74a7a6d893f132eaf98d9d6601R1-R23): Added a corresponding test method to verify that `CollectionView2` can find the ancestor type and handle the binding correctly.

### Issues Fixed

Fixes #26066

rmarinho committed:
[testing] Add repo for issue #26066

rmarinho committed:
[iOS] Make sure to call disconnect handler

rmarinho committed:
[iOS] TemplatedCell2 should be added as logical child

rmarinho committed:
[testing] Add test for issue #26066

mattleibow commented on src/Controls/src/Core/Handlers/Items2/iOS/ItemsViewController2.cs:
```diff
@@ -410,7 +410,7 @@ internal void UpdateView(object view, DataTemplate viewTemplate, ref UIView uiVi
 				uiView?.Dispose();
 				uiView = null;
-
+				formsElement?.Handler?.DisconnectHandler();
```
Do we need the dispose above? doesn't the disconnect do it? And also, @PureWeen said dispose is the evil path to crashes.

PureWeen replied:
We should remove the `Dispose`

We never want to call `Dispose` on anything. 
We should also make sure all the code that's inside any `Dispose` methods related to CV get moved to a disconnect path

PureWeen replied:
https://github.com/dotnet/maui/issues/26134

rmarinho replied:
We can try move the Dipose calls all in another PR for #26134

mattleibow commented on src/Controls/src/Core/Handlers/Items2/iOS/TemplatedCell2.cs:
```diff
@@ -107,49 +108,42 @@ public override void PrepareForReuse()
 
 		public void Bind(View virtualView, ItemsView itemsView)
+		{
+			BindVirtualView(virtualView, itemsView.BindingContext, itemsView, false);
```
Is this `itemsView.BindingContext` correct? Will that be bindng an item to the whole VS and not the item object?

mattleibow replied:
I suppose that is what the existing code was doing, but just checking to see if the OG code was correct.

rmarinho replied:
Yeah this is for the Headers, so basically if you don t give a template and just give a plain headerView or footerview. I just refactored the code here

rmarinho commented on src/Controls/src/Core/Handlers/Items2/iOS/TemplatedCell2.cs:
```diff
@@ -107,49 +108,42 @@ public override void PrepareForReuse()
+			BindVirtualView(virtualView, itemsView.BindingContext, itemsView, false);
+		}
+
+		private void BindVirtualView(View virtualView, object bindingContext, ItemsView itemsView, bool needsContainer)
```
drop private hahahah

rmarinho committed:
[iOS] We just need container when is not template

[iOS] Remove dispose

[ios] Revert to view.SetValueFromRenderer

Revert dispose changes

Fix copy

Add CV1 and CV2 hardcoded

Revert "Revert dispose changes"

This reverts commit 94affddcd8c4596fb891ebe9983a32d46947794e.

Cleanup
